builtin/docker/registry.go

```diff
@@ -43,102 +33,50 @@ func (r *Registry) Push(
 	ui terminal.UI,
 	log hclog.Logger,
 ) (*Image, error) {
-	stdout, _, err := ui.OutputWriters()
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition, "unable to create output for logs:%s", err)
-	}
-
-	sg := ui.StepGroup()
-	step := sg.Add("Initializing Docker client...")
-	defer func() { step.Abort() }()
-
-	cli, err := wpdockerclient.NewClientWithOpts(client.FromEnv)
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition, "unable to create Docker client:%s", err)
-	}
-	cli.NegotiateAPIVersion(ctx)
-
-	step.Update("Tagging Docker image: %s => %s:%s", img.Name(), r.config.Image, r.config.Tag)
-
-	target := &Image{Image: r.config.Image, Tag: r.config.Tag}
-	err = cli.ImageTag(ctx, img.Name(), target.Name())
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "unable to tag image:%s", err)
+	target := &Image{
+		Image: r.config.Image,
+		Tag:   r.config.Tag,
 	}
-
-	step.Done()
-
-	if r.config.Local {
-		return target, nil
+	if !r.config.Local {
+		target.Location = &Image_Registry{Registry: &empty.Empty{}}
```

Comment:
Why the logic change from the previous impl? Before we'd just return the image not doing anything. I think this will break various folks that had set the local flag.

Reply:
This shouldn't break anyone since the `Location` field is new and didn't exist before. 

But I feel like the `Location` field should represent where the image is known to be. If we aren't pushing it to a remote repository, I think we should just retain what `Location` was previously set. Thoughts?